packages/eth-json-rpc-provider/src/safe-event-emitter-provider.ts

```diff
@@ -35,7 +35,7 @@ export function convertEip1193RequestToJsonRpcRequest<
     id = uuidV4(),
     jsonrpc = '2.0',
     method,
-    params = {},
+    params = [],
```

Comment:
Have you considered omitting default params instead? It's optional in the EIP-1193 spec.

Reply:
I am not sure why that type error would happen, the property is marked as optional 🤔

Reply:
> params: Params | undefined; is not assignable to the Params | Record<never, never>.

Hmm, could this happen because we still return an object with a `params` key which will have an undefined value, instead of skipping the params key altogether?

Reply:
I tried getting past this type error locally but was unable to understand the cause. It seems that there is some conflict between the types of `eip1193Request.params` and the return value `params` property. Even when I try omitting `params` when it's unset, I get this error:

```
Types of property 'params' are incompatible.
    Type 'Params' is not assignable to type '((Record<string, Json> | Json[]) & ExactOptionalGuard) & Params'.
      Type 'JsonRpcParams' is not assignable to type '((Record<string, Json> | Json[]) & ExactOptionalGuard) & Params'.
        Type 'Json[]' is not assignable to type '((Record<string, Json> | Json[]) & ExactOptionalGuard) & Params'.
          Type 'Json[]' is not assignable to type 'Json[] & ExactOptionalGuard & Params'.
            Type 'Params' is not assignable to type 'Json[] & ExactOptionalGuard & Params'.
              Type 'JsonRpcParams' is not assignable to type 'Json[] & ExactOptionalGuard & Params'.
                Type 'Json[]' is not assignable to type 'Json[] & ExactOptionalGuard & Params'.
                  Type 'Json[]' is not assignable to type 'Params'.
                    'Json[]' is assignable to the constraint of type 'Params', but 'Params' could be instantiated with a different subtype of constraint 'JsonRpcParams'.
```

Seems like a TypeScript bug. The two values have the same type, `Params`, but it's resolving one to the type constraint then saying it's not narrow enough to match `Params` 😓 .

We can workaround with a type case in the meantime though.

Reply:
Looking back at how this worked prior to #4422, we had previously omitted `params` if they were missing from the request. We had never assigned any default, until the change was introduced in #4422 that caused this bug. Continuing to omit `params` seems safer, less likely to break something.

Even if we default to `[]`, there may be some RPC endpoints that treat that as an invalid request if a method doesn't accept any params.

Reply:
That is what I understood @mikesposito 's suggestion to be; that's what I tried locally when I encountered the type error that I posted.

My most recent comment had a typo in it, apologies for the confusion, it is now edited. My most recent suggestion was to use a type cast to get around the type error.

Though probably we should use `// ts-expect-error` instead of a cast, and explain the error to the best of our ability, and why we cannot workaround it.

Reply:
The suggestion does work, but only due to a pre-existing invalid type that I had fixed locally when testing. Details here: https://github.com/MetaMask/core/pull/4603#discussion_r1715378903